### DISCUSSION TITLE
Reverse engineering challenge idea

### DISCUSSION THREAD

**Discussion by zVSciy** [CTF-FlagFrenzy/challenges#7] [Ideas]
## Python Layer: Dynamic XOR Encryption and Hashing

The Python layer handles the first stage of encryption and hides crucial parts of the flag generation. The encryption is based on dynamic XOR encryption with a runtime-generated key that depends on certain system parameters. The goal of this layer is to make the flag difficult to access through a combination of `XOR` and `hashing`.

## JavaScript Layer: Obfuscated Closures and Transformation

The JavaScript layer is the second stage of encryption. This layer is designed to confuse participants through obfuscation and closures. Here, the flag is further obfuscated through a transformation (e.g., `Base64`) and hidden within functions that are difficult to trace.

@KaveXD 
@IlariaBrown 
@kingdanxi

**Comment by KaveXD** [comment]
> The goal of this layer is to make the flag difficult to access through a combination of XOR and hashing.  

Hashing is practically irreversible, XOR is also irreversible without having the binary stream that was used (which we don't if this is generated at runtime)
